PSReadLine/UndoRedo.cs

```diff
@@ -59,19 +59,13 @@ private void SaveEditItem(EditItem editItem)
             _undoEditIndex = _edits.Count;
         }
 
-        private void StartEditGroup()
+        internal void StartEditGroup()
         {
-            if (_editGroupStart != -1)
-            {
-                // Nesting not supported.
-                throw new InvalidOperationException();
-            }
```

Comment:
If I understand correctly, `_groupUndoHelper` and `_groupUndoStates` are used only when in VI mode, and the starting nested edit group issue is only relevant in VI modes.
If so, then maybe we should only skip the `_editGroupStart != -1` check in `GroupUndoHelper.StartGroup`, but not skipping generally in `StartEditGroup`.

Reply:
I have re-introduced a protection in the general `StartEditGroup()` method.

Reply:
Sorry for the confusion.

Taking a step back and re-assessing the original feedback, I’m not actually in favor of the requested change.
My understanding is that this would allow nested (pending) edit groups in VI mode only, but still protected the general case from unwantingly starting nested groups.

I have pushed a last commit to address this feedback but this leads to code that is – in my opinion – not very elegant.

Please, let me know what you think.